winter/json_encoder.py

```diff
@@ -1,13 +1,120 @@
+import datetime
+import decimal
+import inspect
+import json
+import uuid
 from enum import Enum
+from typing import Callable
+from typing import Dict
+from typing import Type
 
-from rest_framework.utils.encoders import JSONEncoder
+__all__ = (
+    'JSONEncoder',
+    'register_encoder',
+)
 
+_encoder_map: Dict[Type, Callable] = {}
+
+
+class JSONEncoder(json.JSONEncoder):
 
-class WinterJSONEncoder(JSONEncoder):
-    """
-    JSONEncoder subclass that knows how to encode enums
-    """
     def default(self, obj):
+
+        func = _encoder_map.get(type(obj))
+
+        if func is not None:
+            return func(obj)
+
         if isinstance(obj, Enum):
```

Comment:
Improve the mechanism to allow handling such cases with register_encoder decorator as well.